src/main/java/shef/servlets/UserServlet.java

```diff
@@ -102,31 +100,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // Get properties from the request.
     Map<String, String[]> parameterMap = request.getParameterMap();
-    Set entrySet = parameterMap.entrySet();
-    Iterator it = entrySet.iterator();
 
     Key userKey = KeyFactory.createKey("User", id);
     String keyString = KeyFactory.keyToString(userKey);
     Entity user;
-    String redirectUrl;
+    String redirectUrl = "/profile-page.html?key=" + keyString;
 
     try {
       // Get existing user with the key.
       user = datastore.get(userKey);
-      redirectUrl = "/profile-page.html?key=" + keyString;
     } catch (EntityNotFoundException e) {
       // Create a new User entity with data from the request.
       user = new Entity(userKey);
       redirectUrl = "/account-creation-finish.html";
     }
 
-    // These properties don't come from the request, so we add them here.
+    // These properties don't come from the request, so we add to the map here.
     user.setProperty("email", email);
     if(profilePicUrl != null) {
       user.setProperty("profile-pic-url", profilePicUrl);
     }
 
-    Entity finalUser = user.clone();
+    final Entity finalUser = user;
```

Comment:
Oh, I meant if you even needed a finalUser, i.e. if you could just modify "user" variable and put it in datastore. If that is not possible, then please revert to cloning.

Reply:
Oh, yeah, I think I need to have this finalUser Entity. I will revert to cloning.

Reply:
Why do I need to revert it though?